src/LineageTree/lineageTree.py

```diff
@@ -2400,6 +2403,445 @@ def get_cells_at_t_from_root(self, r: [int, list], t: int = None) -> list:
+    ) -> (float, plt.figure):
+        """
+        Plots DTW trajectories aligment between two cell cycles in 2D or 3D
+
+            Args:
+                nodes1 (int): node to compare distance
+                nodes2 (int): node to compare distance
+                threshold (int): set a maximum number of points a track can have
+                regist (boolean): Rotate and translate trajectories
+                start_d (int): start delay
+                back_d (int): end delay
+                w (int): window size
+                fast (boolean): True if the user wants to run the fast algorithm with window restrains
+                centered_band (boolean): if running the fast algorithm, True if the windown is centered
+                three_dim (boolean): True if plot is 3D, else 2D
+                alig (boolean): True to show alignment on plot
+
+            Returns:
+                (float) DTW distance
+                (figue) Trajectories Plot
+        """
+        distance, alignment, cost_mat, pos_cycle1, pos_cycle2 = self.calculate_dtw(
+            nodes1,
+            nodes2,
+            threshold,
+            regist,
+            start_d,
+            back_d,
+            fast,
+            w,
+            centered_band,
+            cost_mat_p=True,
+        )
+
+        fig = plt.figure(figsize=(10, 6))
+        if three_dim:
+            ax = fig.add_subplot(1, 1, 1, projection="3d")
+        else:
+            ax = fig.add_subplot(1, 1, 1)
+
+        if three_dim:
+            ax.plot(
+                pos_cycle1[:, 0],
+                pos_cycle1[:, 1],
+                pos_cycle1[:, 2],
+                "-",
+                label=f"root = {nodes1}",
+            )
+            ax.plot(
+                pos_cycle2[:, 0],
+                pos_cycle2[:, 1],
+                pos_cycle2[:, 2],
+                "-",
+                label=f"root = {nodes2}",
+            )
+        else:
+            ax.plot(
+                pos_cycle1[:, 0],
+                pos_cycle1[:, 1],
+                "-",
+                label=f"root = {nodes1}",
+            )
+            ax.plot(
+                pos_cycle2[:, 0],
+                pos_cycle2[:, 1],
+                "-",
+                label=f"root = {nodes2}",
+            )
+
+        connections = [[pos_cycle1[i], pos_cycle2[j]] for i, j in alignment]
+
+        for connection in connections:
+            xyz1 = connection[0]
+            xyz2 = connection[1]
+            x_pos = [xyz1[0], xyz2[0]]
+            y_pos = [xyz1[1], xyz2[1]]
+            z_pos = [xyz1[2], xyz2[2]]
+
+            if alig:
+                if three_dim:
+                    ax.plot(x_pos, y_pos, z_pos, "k--", color="grey")
+                else:
+                    ax.plot(x_pos, y_pos, "k--", color="grey")
+
+        ax.set_aspect("equal")
+        ax.set_xlabel("x position")
+        ax.legend()
+        ax.set_ylabel("y position")
+        fig.tight_layout()
+
+        return distance, fig
```

Comment:
the alignment doesn't work for the pca projection, so I added this warning

Reply:
Fair enough. You should use the warning then instead of print